### PR TITLE
ENT-4313: Fixed memory leak in mustache rendering

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1770/)

https://ci.cfengine.com/job/pr-pipeline/1770/

File: cf-agent/verify_files.c [hunk 1]
```diff
@@ -608,9 +608,12 @@ static PromiseResult RenderTemplateMustache(EvalContext *ctx, const Promise *pp,
     Attributes a = *attr; // TODO: Try to remove this copy
     PromiseResult result = PROMISE_RESULT_NOOP;
 
+    JsonElement *destroy_this = NULL;
```
Comment (vpodzime) [inline cf-agent/verify_files.c]: Cool name

Comment (olehermanse) [reply cf-agent/verify_files.c]: Yes. The ownership of these Attributes structs is really bad, but requires a lot more effort to fix.

**Comment by olehermanse** [comment]
Memory leak introduced in https://github.com/cfengine/core/pull/2941

(This is actually correct in 3.10, and that PR removed the `JSONDestroy`.)

**Comment by olehermanse** [comment]
Cherry-pick:
https://github.com/cfengine/core/pull/3462

**Pull request merged**